src/reflect/scala/reflect/internal/tpe/FindMembers.scala

```diff
@@ -142,11 +161,11 @@ trait FindMembers {
              !isPrivateLocal // private[this] only a member from within the selector class. (Optimization only? Does the spec back this up?)
           && (
                   !seenFirstNonRefinementClass
-               || refinementParents.contains(owner)
+               || isParentOfAnyRefinementClass(owner)
              )
         )
 
-      (!isPrivate || admitPrivate(sym)) && (sym.name != nme.CONSTRUCTOR || owner == initBaseClasses.head)
+      (sym.name != nme.CONSTRUCTOR || owner == initBaseClasses.head) && (!isPrivate || (selectorClass == owner) || admitPrivate(sym))
```

Comment:
What was the reason for changing the order of the checks here?

Reply:
The reason was to make use of the short-circuiting of boolean operators, so as to delay the evaluation of the call to `admitPrivate` method until it is strictly needed.